.travis.yml

```diff
@@ -14,24 +14,29 @@ cache:
   timeout: 600
   directories:
     - $HOME/cache
-
 jobs:
   include:
     - stage: build third party
       script:
         - cd earth_enterprise/src
-        - if [ -f $HOME/cache/third_party.tgz ]; then tar xf $HOME/cache/third_party.tgz; fi
-        - scons -j3 release=1 third_party > build.log
+        - if [ -f $HOME/cache/third_party$CPP_STD.tgz ]; then tar xf $HOME/cache/third_party$CPP_STD.tgz; fi
+        - scons -j3 release=1 cpp_standard=gnu++$CPP_STD third_party > build.log
         - mkdir -p $HOME/cache
-        - tar cfz $HOME/cache/third_party.tgz NATIVE-REL-x86_64 .sconsign.dblite .sconf_temp
+        - tar cfz $HOME/cache/third_party$CPP_STD.tgz NATIVE-REL-x86_64 .sconsign.dblite .sconf_temp
+      env:
+        - CPP_STD=98
```

Comment:
This seems to just be setting the CPP_STD variable twice in the same job. Maybe try adding a matrix section above jobs:

```
matrix:
  include:
    -env:
         - CPP_STD=98
    -env:
         - CPP_STD=11
```
Example: https://docs.travis-ci.com/user/languages/cpp/#GCC-on-Linux

Reply:
Another option might be to have 2 env sections under the stage:

```
jobs:
   include:
     - stage: build third party
       script:
         ...
       - env:
         CPP_STD=98
       - env:
         CPP_STD=11
```

Reply:
I'll give that a try.  The documentation doesn't help much with what I want to do.  I think I can get it working the way I want if you duplicate the script code twice and name the stages the same.  I duplicated the stages but called them something different but that made the run take twice as long....  I saw an example where they named the stage the same and liked named stages run in parallel (at least that is what the example said).  But I didn't like that because I wanted to re-use the code in the stages....

Reply:
Turns out that isn't valid yaml syntax.  I found https://www.json2yaml.com/ so I can do things in JSON and see what the equivalent yaml is.  I think what I want to do can not be done with the new stage syntax.  I basically want to matrix the job object to run with different environments but it looks like that isn't supported...  Maybe if I move the env array up one more level...  If that doesn't work I'm going to just go with repeating the stages...